flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java

```diff
@@ -152,10 +119,9 @@ public ArrayKeySelector(int... fields) {
 
 		@Override
 		public Tuple getKey(IN value) throws Exception {
-			key = (Tuple) tupleClasses[fields.length - 1].newInstance();
+			Tuple key = Tuple.getTupleClass(fields.length).newInstance();
 			for (int i = 0; i < fields.length; i++) {
-				int pos = fields[i];
```

Comment:
How far are we with the release candidate? We might still be able to include it.